hazelcast/src/main/java/com/hazelcast/internal/probing/ProbeRegistry.java

```diff
@@ -0,0 +1,107 @@
+package com.hazelcast.internal.probing;
+
+import com.hazelcast.internal.metrics.Probe;
+import com.hazelcast.internal.metrics.ProbeLevel;
+import com.hazelcast.spi.annotation.Beta;
+import com.hazelcast.spi.annotation.PrivateApi;
+import com.hazelcast.spi.impl.servicemanager.ServiceManager;
+
+/**
+ * A service made accessible to core services so they have a chance to register
+ * the "root objects" of the system.
+ * 
+ * After the registration phase the set of this {@link ProbeSource}s is fix.
+ * There is no need to unregister them later on as they always have the choice
+ * to stop rendering something.
+ * 
+ * Metrics about things that change over time require a corresponding
+ * {@link ProbeSource} (existing since the startup) that does probe the
+ * available instances as they appear and stops doing that as they disappear.
+ */
+@Beta
+@PrivateApi
+public interface ProbeRegistry {
+
+    /**
+     * Called once at startup, typically by a core service registering itself.
+     * 
+     * @param source a object that "knows" how to render metrics in its context
+     */
+    void register(ProbeSource source);
+
+    /**
```

Comment:
Why not drop the method below and only offer the register(Object method)? Makes the API easier to use; don't see the added value of 2 methods that are nearly identical.

Reply:
This adds type safety where we can have it. This helps finding cleanups that need to be done after changes where e.g. we change what class implements `ProbeSource`. I just added `registerIfSource` because of a few cases where we don't know because of the way plugins work.

Reply:
ok